sycl/include/sycl/ext/intel/esimd/detail/simd_obj_impl.hpp

```diff
@@ -318,13 +318,31 @@ class [[__sycl_detail__::__uses_aspects__(
   /// @tparam Flags Specifies memory address alignment. Affects efficiency of
   ///   the generated code.
   /// @param ptr The memory address to read from.
-  template <typename Flags = element_aligned_tag,
-            typename = std::enable_if_t<is_simd_flag_type_v<Flags>>>
-  simd_obj_impl(const Ty *ptr, Flags = {}) noexcept {
+  template <typename Flags,
+            typename std::enable_if_t<is_simd_flag_type_v<Flags>, bool> = true>
```

Comment:
unfortunately the old style doesn't work because we have two overloads with default arguments with templated types. the problem is can be seen here:
https://godbolt.org/z/4nEx74j6e


The suggested workaround is to make the original parameter a type parameter:
https://stackoverflow.com/questions/51292574/sfinae-constructors

Reply:
I see, thank you for the details.